plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -231,3 +608,1407 @@ def set_wins_registration(self, mgrattr):
+                    "msg": "Unsupported input parameters: %s" % str(list(set(raid.keys()) - set(input_list)))
+                }
+
+            if not_defined:
+                return {
+                    "ret": False,
+                    "changed": False,
+                    "msg": "Input parameters %s should not be empty" % (str(not_defined))
+                }
+
+            return {
+                "ret": True,
+                "changed": False,
+                "msg": "Input parameters verified"
+            }
+
+    def verify_input_paramters(self, raid_data, command="CreateLogicalDrives"):
+        # Verifying input parameters passed to the CreateLogicalDrives and
+        # CreateLogicalDrivesWithPArticularPhysicalDrives modules
+        if command == "CreateLogicalDrives":
+            input_list = ['LogicalDriveName', 'Raid', 'DataDrives']
+            drive_input = ['DataDriveCount', 'DataDriveMediaType',
+                           'DataDriveInterfaceType', 'DataDriveMinimumSizeGiB']
+            for raid in raid_data:
+                missing_param = []
+                not_defined = []
+                for input in input_list:
+                    if input not in raid.keys():
+                        missing_param.append(input)
+                    elif not raid[input]:
+                        not_defined.append(input)
+
+                if 'DataDrives' not in raid.keys():
+                    missing_param = missing_param + drive_input
+                else:
+                    for drive in drive_input:
+                        if drive not in raid["DataDrives"]:
+                            missing_param.append(drive)
+                        elif drive != "DataDriveMinimumSizeGiB" and not raid["DataDrives"][drive]:
+                            not_defined.append(drive)
+                        elif drive == "DataDriveMinimumSizeGiB" and \
+                                not raid["DataDrives"]["DataDriveMinimumSizeGiB"] and \
+                                raid["DataDrives"]["DataDriveMinimumSizeGiB"] != 0:
+                            not_defined.append(drive)
+
+            return self.validation_error(raid, input_list, missing_param, not_defined, drive_input)
+
+        elif command == "CreateLogicalDrivesWithParticularPhysicalDrives":
+            input_list = ['LogicalDriveName', 'CapacityGB', 'Raid', 'DataDrives']
+
+            for raid in raid_data:
+                missing_param = []
+                not_defined = []
+                for input in input_list:
+                    if input not in raid.keys():
+                        missing_param.append(input)
+                    elif not raid[input]:
+                        not_defined.append(input)
+
+            return self.validation_error(raid, input_list, missing_param, not_defined, [], command="CreateLogicalDrivesWithParticularPhysicalDrives")
+
+    def check_physical_drives(self, raid_data, unused_physical_drives, command="CreateLogicalDrives"):
```

Comment:
Can this be renamed for Smart Storage to avoid confusion with generic drive checking?

Reply:
We've renamed all the commands in the other PRs raised to include SmartStorage in the name. 
Ex: GetLogicalDrives to GetSmartStorageLogicalDrives. [5540](https://github.com/ansible-collections/community.general/pull/5540/)
Do we need to rename all the functions being called as well?

Reply:
Yes, since they're functions of the module, and other developers might be misled as to what their intent is.

Reply:
The other concern is if we grow the standard module to include more storage-related functions, this name could conflict with what we'd like to do there.

Reply:
All functions have now been renamed. Resolving this conversation. Thanks!